.github/workflows/ci.yml

```diff
@@ -145,6 +145,89 @@ jobs:
           gcloud alpha storage cp "${BUILD_DIR_ARCHIVE}" "${GCS_ARTIFACT}"
           echo "::set-output name=gcs-artifact::${GCS_ARTIFACT}"
 
+  build_tf_integrations:
+    runs-on:
+      # Hacks, and order matters. See the comment at the top of the file.
+      - self-hosted
+      - runner-group=${{ github.event_name == 'pull_request' && 'presubmit' || 'postsubmit' }}
+      - cpu
+      - os-family=Linux
+    outputs:
+      binaries-dir: ${{ steps.build.outputs.binaries-dir }}
+      binaries-archive: ${{ steps.archive.outputs.binaries-archive }}
+      gcs-artifact: ${{ steps.upload.outputs.gcs-artifact }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          submodules: true
+      - name: "Building TF binaries"
+        id: build
+        env:
+          IREE_TF_BINARIES_OUTPUT_DIR: iree-tf-binaries
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+            --env "IREE_TF_BINARIES_OUTPUT_DIR=$(realpath "${IREE_TF_BINARIES_OUTPUT_DIR}")" \
+            gcr.io/iree-oss/frontends-swiftshader@sha256:41e516b8c1b432e3c02896c4bf4b7f06df6a67371aa167b88767b8d4d2018ea6 \
+            build_tools/cmake/build_tf_binaries.sh
+          echo "::set-output name=binaries-dir::${IREE_TF_BINARIES_OUTPUT_DIR}"
+      - name: "Creating archive of binaries"
+        id: archive
+        env:
+          BINARIES_ARCHIVE: tf-binaries.tar
+          BINARIES_DIR: ${{ steps.build.outputs.binaries-dir }}
+        run: |
+          tar -cf "${BINARIES_ARCHIVE}" "${BINARIES_DIR}"
+          echo "::set-output name=binaries-archive::${BINARIES_ARCHIVE}"
+      - name: "Uploading binaries archive"
+        id: upload
+        env:
+          GCS_ARTIFACT: ${{ env.BUCKET }}/${{ github.sha }}/${{ steps.archive.outputs.binaries-archive }}
+          BINARIES_ARCHIVE: ${{ steps.archive.outputs.binaries-archive }}
+        run: |
+          gcloud alpha storage cp "${BINARIES_ARCHIVE}" "${GCS_ARTIFACT}"
+          echo "::set-output name=gcs-artifact::${GCS_ARTIFACT}"
+
+  test_tf_integrations:
```

Comment:
nit: how do you feel about ordering in this file? While you have `test_tf_integrations` right after `build_tf_integrations`, I'd also consider putting it after `test_all` (list the simplest tests first)

Reply:
Yeah I was recognizing that the ordering was weird here and then didn't change it. I guess sorting topologically is probably good and then simplest->most complicated seems reasonable?

Reply:
> I guess sorting topologically is probably good and then simplest->most complicated seems reasonable?

SGTM.